Lib/inspect.py

```diff
@@ -1213,6 +1213,34 @@ def getblock(lines):
             pass
     return lines[:blockfinder.last]
 
+def _get_generator_expression_code_source(object, lines):
+    """Get source lines for a generator expression code object
+
+    The returned lines contain only the code for the generator expression,
+    so if two generator expressions are defined on the same line then only
+    the source for this particular object will be returned.
+    Anything after the generator expression is removed and anything on the first
+    line before the generator expression is replaced with whitespace.
+    """
+
+    # get the last entry in co_positions that doesn't have None for column numbers
+    code_position = None
+    for pos in object.co_positions():
+        if pos[2] is not None and pos[3] is not None:
+            code_position = pos
+
+    # co_positions are all None, likely because CLI flag "-X no_debug_ranges" is set.
+    if code_position is None:
+        raise OSError('Cannot get source for generator expression code object without debug ranges')
```

Comment:
There should definitely not be an `OSError`. `OSError` is used for errors that rely on the OS API or similar. I'd recommend using `ValueError` instead.

Reply:
I thought it a strange choice too, but `OSError` is used throughout `insepct` for the situtation that the source can't be found, including in the docstring and docs of `inspect.getsource` itself as what the user can expect to be raised. 

```python
def getsource(object):
    """Return the text of the source code for an object.

    The argument may be a module, class, method, function, traceback, frame,
    or code object.  The source code is returned as a single string.  An
    OSError is raised if the source code cannot be retrieved."""
    lines, lnum = getsourcelines(object)
    return ''.join(lines)
```

Reply:
You're right. Let's keep `OSError` then.

Reply:
Perhaps an idea to raise a separate issue to make the change module wide, but potentially difficult from a backwards-compatability perspective.

Reply:
> Perhaps an idea to raise a separate issue to make the change module wide, but potentially difficult from a backwards-compatability perspective.

I think there's no chance to change it from OSError to any other exception. These usages of OSError are just semantically wrong for me. From the point of view of users of inspect, I guess there are no problems with handling OSError instead of handling any other exception.